Assets/Scripts/Player/PlayerController.cs

```diff
@@ -111,8 +118,63 @@ private void Update()
             Vector3 playerPosition = transform.position;
             _camera.transform.position = playerPosition + _cameraPosition;
             _camera.transform.LookAt(playerPosition);
+
+            // Check and makes character facing the target object when moving toward it
+            if (target != null) FaceTarget();
+        }
+
+        /// <summary>
+        /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+        /// (Won't conflict with character attacking enemy or moving on the map)
+        /// </summary>
+        #region Interactables Functions
+        
+        // This will make character moving to the target object
+        public void MovingToTarget(Interactable newTarget)
+        {
+            _navMeshAgent.stoppingDistance = newTarget.radius * .8f;
+            _navMeshAgent.updateRotation = false;
+            target = newTarget.transform;
+        }
+
+        // Makes character stop moving to target object
+        public void StopMovingToTarget()
```

Comment:
The nav mesh agent should stop automatically if it's close enough to the target. Changing the stopping distance manually isn't the best idea.

Reply:
This is a "naming" mistake from me. Actually this will make the character not targeting the object anymore and change the stopping distance to 0 so the character can move all the way to where the mouse click without stopping at a distance.